runtime/src/main/java/org/corfudb/runtime/view/AddressSpaceView.java

```diff
@@ -148,11 +152,21 @@ public void write(@Nonnull IToken token, @Nonnull Object data, @Nonnull CacheOpt
                 l.getReplicationMode(token.getTokenValue())
                         .getReplicationProtocol(runtime)
                         .write(e, ld);
-            } catch (OverwriteException | WriteSizeException ex) {
-                // If we have an Overwrite exception, it is already too late for trying
-                // to validate the state of the write, we know that the write didn't complete.
-                // Large writes are also rejected right away.
-                throw ex;
+            } catch (OverwriteException ex) {
```

Comment:
If you have a chain longer than the number of retries, would it be possible for the transaction to fail even though the right went through?